src/dbally/collection.py

```diff
@@ -208,12 +221,18 @@ async def ask(self, question: str, dry_run: bool = False, return_natural_respons
             event_tracker=event_tracker,
             n_retries=self.n_retries,
             dry_run=dry_run,
+            llm_options=llm_options,
         )
         end_time_view = time.monotonic()
 
         textual_response = None
         if not dry_run and return_natural_response:
-            textual_response = await self._nl_responder.generate_response(view_result, question, event_tracker)
+            textual_response = await self._nl_responder.generate_response(
```

Comment:
Maybe this is overthinking but I imagine such a scenario, in which I want to use different LLM for View selector, collection and nl_responder. To exemplify nl_responder gets a lot of input tokens but doesn't have to perform complicated "thinking" skills, so I can use smaller LLM and save some money. 

Now the question is whether such an approach will not lead to problems when different clients are used within different blocks.

Reply:
I think this is over-thinking it at this stage, we don't have the logic for many different LLM clients yet. In any case, in the future we can just put more parameters to parameterize each step separately.